src/main/java/org/jahia/modules/appshell/Main.java

```diff
@@ -100,6 +100,11 @@ public String getContextPath() {
             };
             // lookup for site
             JahiaSite site = ServicesRegistry.getInstance().getJahiaSitesService().getDefaultSite(JCRSessionFactory.getInstance().getCurrentUserSession());
+
+            if (site == null) {
+                site = ServicesRegistry.getInstance().getJahiaSitesService().getFirstSiteFound(JCRSessionFactory.getInstance().getCurrentUserSession(), "systemsite");
+            }
+
             // use system site if no site readable.
             if (site == null) {
                 site = ServicesRegistry.getInstance().getJahiaSitesService().getSiteByKey("systemsite");
```

Comment:
Would this block still reachable or useful?

Reply:
yes, because the previous one excludes systemsite

Reply:
Oh. I did not check the documentation and assumed that the last parameter was a "default value".